### PR TITLE
add class for button-link

### PR DESCRIPTION
**Pull request by jseppi**
Adds a new `.button-link` class for buttons to make them look like links. This has precedent in other frameworks, like Bootstrap.

It will be useful for the Cancel button styling described in https://github.com/18F/calc/issues/651

Screenshot from Style Guide:

screen shot 2016-09-05 at 3 38 08 pm

### CODE REVIEW

File: frontend/source/sass/components/_buttons.scss [hunk 1]
```diff
@@ -86,3 +86,34 @@ input.button-secondary[type="button"] {
...
 }
+
+button.button-link,
+input.button-link[type="submit"],
+input.button-link[type="reset"],
+input.button-link[type="button"] {
+
+  border: none;
+  padding-top: 0;
+  padding-bottom: 0;
```
Comment (jseppi) [inline frontend/source/sass/components/_buttons.scss]: I left the left and padding alone, but I could also see just setting `padding: 0;` here.

**Comment by toolness** [comment]
Huh, it feels a bit odd to make something that looks like a link but changes server state (and potentially even loses user data), but I suppose it's no less weird than making a link that looks like a button. 😁

**Comment by toolness** [comment]
Would it be helpful to use [`appearance: none`](https://css-tricks.com/almanac/properties/a/appearance/) here for browsers that support it, just to make sure other weird native platform CSS properties don't bleed through?

**Comment by jseppi** [comment]
Neat, didn't know about `appearance`!

**Comment by jseppi** [comment]
Added `appearance` rules. Will wait for 👍 from @hbillings.

**Comment by hbillings** [comment]
LGTM!

**Pull request merged**